src/goto-symex/goto_symex_state.cpp

```diff
@@ -200,7 +200,12 @@ void goto_symex_statet::assignment(
   // for value propagation -- the RHS is L2
 
   if(!is_shared && record_value && goto_symex_is_constantt()(rhs))
-    propagation[l1_identifier] = rhs;
+  {
+    if(propagation.has_key(l1_identifier))
+      propagation.replace(l1_identifier, rhs);
```

Comment:
Check if `rhs` is different from the value already in the map (see above).

Reply:
Done.